api/src/main/java/marquez/db/RunDao.java

```diff
@@ -140,54 +140,80 @@ LEFT OUTER JOIN (
 
   @SqlQuery(
       """
-          SELECT r.*, ra.args, f.facets,
-          j.namespace_name, j.name, jv.version AS job_version,
-          ri.input_versions, ro.output_versions, df.dataset_facets
-          FROM runs_view AS r
-          INNER JOIN jobs_view j ON r.job_uuid=j.uuid
-          LEFT JOIN LATERAL
-          (
-            SELECT rf.run_uuid, JSON_AGG(rf.facet ORDER BY rf.lineage_event_time ASC) AS facets
-            FROM run_facets_view rf
-            WHERE rf.run_uuid=r.uuid
-            GROUP BY rf.run_uuid
-          ) AS f ON r.uuid=f.run_uuid
-          LEFT OUTER JOIN run_args AS ra ON ra.uuid = r.run_args_uuid
-          LEFT OUTER JOIN job_versions jv ON jv.uuid=r.job_version_uuid
-          LEFT OUTER JOIN (
-           SELECT im.run_uuid, JSON_AGG(json_build_object('namespace', dv.namespace_name,
-                  'name', dv.dataset_name,
-                  'version', dv.version,
-                  'dataset_version_uuid', uuid
-                  )) AS input_versions
-           FROM runs_input_mapping im
-           INNER JOIN dataset_versions dv on im.dataset_version_uuid = dv.uuid
-           GROUP BY im.run_uuid
-          ) ri ON ri.run_uuid=r.uuid
-          LEFT OUTER JOIN (
-            SELECT run_uuid, JSON_AGG(json_build_object('namespace', namespace_name,
-                                                        'name', dataset_name,
-                                                        'version', version,
-                                                        'dataset_version_uuid', uuid
-                                                        )) AS output_versions
-            FROM dataset_versions
+          WITH filtered_jobs AS (
+            SELECT
+                jv.uuid,
+                jv.namespace_name,
+                jv.name
+            FROM jobs_view jv
+            WHERE jv.namespace_name=:namespace AND (jv.name=:jobName OR :jobName = ANY(jv.aliases))
+          ),
+          run_facets_agg AS (
+            SELECT
+                run_uuid,
+                JSON_AGG(facet ORDER BY lineage_event_time ASC) AS facets
+            FROM run_facets_view
+            WHERE
+                run_uuid IN (SELECT uuid FROM runs_view WHERE job_uuid IN (SELECT uuid FROM filtered_jobs))
             GROUP BY run_uuid
-          ) ro ON ro.run_uuid=r.uuid
-          LEFT OUTER JOIN (
+          ),
+          input_versions_agg AS (
+               SELECT
+                   im.run_uuid,
+                   JSON_AGG(json_build_object('namespace', dv.namespace_name,
+                        'name', dv.dataset_name,
+                        'version', dv.version,
+                        'dataset_version_uuid', dv.uuid
+                   )) AS input_versions
+               FROM runs_input_mapping im
+               INNER JOIN dataset_versions dv ON im.dataset_version_uuid = dv.uuid
+               WHERE
```

Comment:
Is this `where .. in (SELECT ...)` necessary to gain performance boost?

Reply:
Ok. Could you then add some comment describing that it's done for this purpose. 

It may be tempting for anyone in future to refactor this code and get rid of this. In such case, our tests will not notice the performance degradation...

Reply:
Done ! ;)